nntrainer/models/neuralnet.cpp

```diff
@@ -375,34 +367,36 @@ sharedConstTensors NeuralNetwork::forwarding(
       node->forwarding(training);
     } else {
       /**
-       currently, it supports FSU asynch mode for inference. The prcedure of
-       FSU is below,
+         currently, it supports FSU asynch mode for inference. The prcedure of
+         FSU is below,
 
-       Prerequests : This function is called node by node at the forwarding
-       function in network graph.
+         Prerequests : This function is called node by node at the forwarding
+         function in network graph.
 
-       Step 1. If the execution order is the first (f==0) then, it will try to
-               load tensors which used at layer 0.
+         Step 1. If the execution order is the first (f==0) then, it will try
+       to load tensors which used at layer 0.
 
-       Step 2. It check whether these tensors from Step 1, then do the
-               forwarding of the first node.
+         Step 2. It check whether these tensors from Step 1, then do the
+                 forwarding of the first node.
 
-       Step 3. Then check the look a head which says how many layer weights need
-               to be loaded before running to hide overehad due to FSU,
+         Step 3. Then check the look a head which says how many layer weights
+       need to be loaded before running to hide overehad due to FSU,
 
-       Step 4. Try to get the tesors by asking tensors for layers which is done
-               by thread pool
+         Step 4. Try to get the tesors by asking tensors for layers which is
+       done by thread pool
 
-       Step 5. Try to release the weights which has execution order less then f.
+         Step 5. Try to release the weights which has execution order less
+       then f.
+
+         Step n. repeat next layer starting with checking the tenosrs are
+       loaded, and if it is loaded, then run forwarding. Every time it
+       finishes the forwarding, ask load tensors for next n layers.
 
-       Step n. repeat next layer starting with checking the tenosrs are loaded,
-               and if it is loaded, then run forwarding. Every time it finishes
-               the forwarding, ask load tensors for next n layers.
       **/
-      auto look_ahead = std::get<props::FsuLookahead>(model_flex_props);
-      model_graph.checkFsuLoadComplete(f);
+      model_graph.checkLoadComplete(f);
       node->forwarding(training);
-      model_graph.LoadTensors(f + look_ahead, look_ahead);
+      model_graph.inActive(f);
+      model_graph.LoadTensors(f + lookahead);
```

Comment:
The current structure is check -> forwarding -> load(current + lookahead). However, this structure seems slightly inefficient.

In my opinion, after forwarding is complete, an inactive call should be made, followed by executing the next load on the inactive memory (without directly calling load). Wouldn't this structure be simpler and more efficient?

To operate in this manner, the manager needs to be modified to run and manage threads.

Reply:
The structure you described seems to be understood as automatically calling the inactive for the corresponding execution order after forwarding ends, and automatically loading this into empty memory.

[ forwarding -> inactive memory]
&& 
[ Manager -> check for inactive memory -> load automatically]

understood correctly.?